src/modules/send_message/index.js

```diff
@@ -0,0 +1,74 @@
+const twitch = require('../../utils/twitch');
+const watcher = require('../../watcher');
+const debug = require('../../utils/debug');
+
+const chatCommands = require('../chat_commands');
+const anonChat = require('../anon_chat');
+const socketClient = require('../../socket-client');
+
+const PATCHED_SENTINEL = () => {};
+
+class SendState {
+    constructor(msg) {
+        this.message = msg;
+        this.defaultPrevented = false;
+    }
+
+    get user() {
+        return twitch.getCurrentUser();
+    }
+
+    preventDefault() {
+        this.defaultPrevented = true;
+        this.message = '';
+    }
+}
+
+let twitchSendMessage;
+const methodList = [
+    // msgObj => tabCompletion.onSendMessage(msgObj),
+    msgObj => chatCommands.onSendMessage(msgObj),
+    msgObj => anonChat.onSendMessage(msgObj)
+];
+
+function bttvSendMessage(channelName, messageToSend) {
+    const channel = twitch.getCurrentChannel();
+    if (channel) {
+        socketClient.broadcastMe(channel.name);
+    }
+
+    const sendState = new SendState(messageToSend);
+
+    for (const method of methodList) {
+        try {
+            method(sendState);
+        } catch (e) {
+            debug.log(e);
+        }
+    }
+
+    if (sendState.defaultPrevented) return;
+    messageToSend = sendState.message;
+    twitchSendMessage.call(this, channelName, messageToSend);
```

Comment:
twitch's function can change. to be safe you should always just take in args and pass through args.. also would be smart to check that messageToSend is a string when pulling it out of args.